rdflib/plugins/parsers/trig.py

```diff
@@ -104,6 +104,8 @@ def graph(self, argstr: str, i: int) -> int:
         self._reason2 = becauseSubGraph
         # type error: Incompatible types in assignment (expression has type "Graph", variable has type "Optional[Formula]")
         self._context = self._store.newGraph(graph)  # type: ignore[assignment]
+        if self._context is not None and self._parentContext is not None:
+            raise SyntaxError("GRAPH may not include a GRAPH " "(#trig-graph-bad-07)")
```

Comment:
```suggestion
            raise SyntaxError("GRAPH may not include a GRAPH")
```

Reply:
This will trigger even for cases where GRAPH is not used, and that is desirable, but in those cases the error message will be a bit misleading.

Also, while I don't exactly condone it, the way that errors are handled for the N3 related parsers is with `self.BadSyntax(...)`, and the `argstr: str, i: int` parameters are passed to it. The benefit of this is that it reports the line number, or at least tries to.

Reply:
Fixed now.